src/Dapr.Actors.AspNetCore/ActorsEndpointRouteBuilderExtensions.cs

```diff
@@ -78,6 +79,12 @@ private static IEndpointConventionBuilder MapActorMethodEndpoint(this IEndpointR
                 var actorId = (string)routeValues["actorId"];
                 var methodName = (string)routeValues["methodName"];
 
+                if (context.Request.Headers.ContainsKey(Constants.ReentrancyRequestHeaderName))
+                {
+                    var daprReentrancyHeader = context.Request.Headers[Constants.ReentrancyRequestHeaderName];
+                    ActorReentrancyContextAccessor.ReentrancyContext = daprReentrancyHeader;
```

Comment:
The other thing we need to do here is **unset** the value after the user-code executes. 

There should be a try/finally around `runtime.Dispatch....` and in the finally, you want to set the context back to null.

Reply:
The **unset** is what fixes where bugs where long-running background tasks started from inside an Actor still think they are running inside an actor.